Assets/Plugins/Source/Editor/Utility/GUIEditorUtility.cs

```diff
@@ -96,47 +97,44 @@ public static void AssigningTextField(string label, ref string value, float labe
             EditorGUIUtility.labelWidth = originalLabelWidth;
         }
 
-        public static void AssigningULongToStringField(string label, ref string value, float labelWidth = -1, string tooltip = null)
+        public static void AssigningULongToStringField(string label, ref ulong? value, float labelWidth = -1, string tooltip = null)
         {
             float originalLabelWidth = EditorGUIUtility.labelWidth;
             if (labelWidth >= 0)
             {
                 EditorGUIUtility.labelWidth = labelWidth;
             }
 
-            try
-            {
-                EditorGUILayout.BeginHorizontal();
-                var newValueAsString = EditorGUILayout.TextField(CreateGUIContent(label, tooltip), value == null ? "" : value, GUILayout.ExpandWidth(true));
+            EditorGUILayout.BeginHorizontal();
 
-                if (GUILayout.Button("Clear", GUILayout.MaxWidth(50)))
-                {
-                    value = null;
-                }
-                else
-                {
-                    if (string.IsNullOrWhiteSpace(newValueAsString))
-                    {
-                        value = null;
-                        return;
-                    }
-
-                    var valueAsLong = ulong.Parse(newValueAsString);
-                    value = valueAsLong.ToString();
-                }
-            }
-            catch (FormatException)
+            var guiLabel = CreateGUIContent(label, tooltip);
+            string textToDisplay = string.Empty;
+            
+            if (value.HasValue)
             {
-                value = null;
+                textToDisplay = value.Value.ToString();
             }
-            catch (OverflowException)
+            
+            string newTextValue = EditorGUILayout.TextField(guiLabel, textToDisplay, GUILayout.ExpandWidth(true));
+            
+            if (GUILayout.Button("Clear", GUILayout.MaxWidth(50)))
```

Comment:
![selectedfield](https://github.com/user-attachments/assets/b1100335-f154-4efb-a6f7-1945b6b1b68f)
This clear button functions correctly in that it eliminates the value. *But*, if you have that value currently selected and then press clear, the value will appear to still be in the config field. Clicking away will reveal it was cleared properly. Is there a way we can make this clear button also clear the selection?

Reply:
Unfortunately, if the field is still selected, then it is considered as something that is still being edited, as in the value is not accessible (or clearable) until focus for the field is lost. Clicking the 'clear' button doesn't move the cursor, so until that happens the value cannot be cleared.